test/viewmodels/new_comment_view_model_integration_test.dart

```diff
@@ -0,0 +1,162 @@
+import "package:cloud_firestore/cloud_firestore.dart";
+import "package:fake_cloud_firestore/fake_cloud_firestore.dart";
+import "package:flutter_test/flutter_test.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/post/post_id_firestore.dart";
+import "package:proxima/models/database/user/user_id_firestore.dart";
+import "package:proxima/models/ui/new_comment_state.dart";
+import "package:proxima/services/database/comment_repository_service.dart";
+import "package:proxima/services/database/firestore_service.dart";
+import "package:proxima/viewmodels/login_view_model.dart";
+import "package:proxima/viewmodels/new_comment_view_model.dart";
+
+import "../mocks/data/comment_data.dart";
+import "../mocks/data/firestore_post.dart";
+import "../mocks/data/firestore_user.dart";
+import "../mocks/data/geopoint.dart";
+
+void main() {
+  group("Testing new comment view model", () {
+    late FakeFirebaseFirestore fakeFirestore;
+    late CommentDataGenerator commentDataGenerator;
+
+    late CommentRepositoryService commentRepository;
+    late AutoDisposeFamilyAsyncNotifierProvider<NewCommentViewModel,
+        NewCommentState, PostIdFirestore> newCommentViewModelProvider;
+
+    late ProviderContainer container;
+
+    late PostIdFirestore postId;
+    late UserIdFirestore userId;
+
+    setUp(() async {
+      fakeFirestore = FakeFirebaseFirestore();
+      commentDataGenerator = CommentDataGenerator();
+      userId = testingUserFirestoreId;
+
+      container = ProviderContainer(
+        overrides: [
+          firestoreProvider.overrideWithValue(fakeFirestore),
+          uidProvider.overrideWithValue(userId),
+        ],
+      );
+
+      commentRepository = container.read(commentRepositoryProvider);
+
+      // Add a post to the database on which the comments will be added
+      final post = FirestorePostGenerator().generatePostAt(userPosition0);
+      await setPostFirestore(post, fakeFirestore);
+      postId = post.id;
+
+      newCommentViewModelProvider = newCommentStateProvider(postId);
+    });
+
+    test("Add a valid comment and reset correctly", () async {
+      final validContent = commentDataGenerator.createMockCommentData().content;
+
+      final newCommentViewModel =
+          container.read(newCommentViewModelProvider.notifier);
+
+      //  --- Check the state before adding the comment
```

Comment:
What is this notation with three dashes ? `// ---`

Do we use this somewhere else in testing ? I don't remember seeing it and don't know what it means.

Reply:
This was to have a more visual separation in the test code.
But I am realising that I didn't used it elsewhere.
Thus I remove it to be consistent with the rest of the codebase,

Done in d235ffe3ed2da2db32217cd8652335ab4fdfd112.